src/event.cpp

```diff
@@ -103,61 +105,156 @@ void serialize_match(const condition_match &match, ddwaf_object &match_map, auto
     ddwaf_object_map_add(&match_map, "parameters", &parameters);
 }
 
-} // namespace
+struct action_tracker {
+    // The blocking action refers to either a block_request or redirect_request
+    // action, the latter having precedence over the former.
+    std::string_view blocking_action{};
+    action_type blocking_action_type{action_type::none};
 
-void event_serializer::serialize(const std::vector<event> &events, ddwaf_result &output) const
+    // Stack trace ID
+    std::string stack_id{};
+
+    std::unordered_set<std::string_view> all{};
```

Comment:
I think it's fair to assume that an `all` within `action_tracker` refers to all actions

Reply:
I find it bizarre to say the least, even just at the linguistic level. "All" not followed by anything means "everything". You mean "all action ids", I imagine, but if you remove "action ids" you have a different word (a pronoun instead of a determiner) that's just spelled the same (think of tudo/todos). Then there's the fact it doesn't convey much information. I imagine that the idea was to contrast with `blocking_action`, but then "all_action_ids" is better. Or would be, because now looking down it seems that the emplace is conditioned on `!is_blocking_action(type)`, so "all" doesn't even seem to mean "all action ids".

Reply:
Well, that last point is fair, I guess a more suitable name would be `all_non_blocking_ids` or something like that.